### ISSUE TITLE
Flip scrren option

### ISSUE THREAD

**Issue by stiegis** [MiSTer-devel/Arcade-Joust2_MiSTer#18]
Would be nice to have a vertical flip option for the game

**Comment by birdybro** [comment]
Yup! I don't know if flip is something that was built-in to the DIP switches on the original arcade system. If not, then a way to flip the graphics sorta like JimmyStones has been doing to some cores will have to be implemented, and it's not too easy to do.

I can however add a flip that only works with the framebuffer like other cores, for now. I was still very early in learning when I ported this core, I'll look into it.

That being said, the framebuffer flip option only works in one of the two orientations. In general that's not a big problem though since I'm sure barely anyone is flipping their modern display upside down randomly.

**Comment by stiegis** [comment]
Hi Bidybro! 

Thanks for giving this a shot. It is not a big deal since I can switch the monitor orientation in a heartbeat with a simple plug for the yoke wires but this is more a matter of convenience. Regards

**Comment by DennisGruendig** [comment]
I would appreciate that too 👍

My CRT does that weird color shift when I turn it counterclockwise, so a flip is the only way, I could play this authentically 😅

**Comment by birdybro** [comment]
@iequalshane told me he was going to look into this, but he is a busy person so no pressure. Just providing a status update from myself for the people wondering here.

**Comment by iequalshane** [comment]
I was hoping to do this without the need of a framebuffer, and I -mostly- succeeded but there's some software timing with tiles that makes it impossible (tiles being written before being drawn in an order that breaks with when you flip everything). I'll check where I left off and see if anything else can be done before giving up and adding a framebuffer.

**Comment by birdybro** [comment]
No worries at all. If it needs a framebuffer we should just use the flip feature built into the framework. I know it's very difficult to add flip/rotate to things like this, it's like modding the arcade hardware on a low level! :)

**Comment by DennisGruendig** [comment]
It‘s good so see that you want to fix it the right way and not just any way 👍

**Comment by bawitdaba** [comment]
> I can however add a flip that only works with the framebuffer like other cores, for now. I was still very early in learning when I ported this core, I'll look into it.
> 
> That being said, the framebuffer flip option only works in one of the two orientations. In general that's not a big problem though since I'm sure barely anyone is flipping their modern display upside down randomly.

My HDMI monitor is mounted +90 degrees rotated so the game is upside down for me :(

I tried custom compiling a version with flip=1 and it was a 45 degree skewed mess. I also tried messing with no_rotate/rotate_ccw and wasn't able to get a 180 flip. Only +90/-90

Is it possible to do a 180 flip (hflip and vflip) in the framebuffer?

**Comment by birdybro** [comment]
I think the added screen rotate might help on your hdmi monitor. Try it out.
